Tella/Data/Database/ServerDatabase.swift

```diff
@@ -162,6 +162,9 @@ extension TellaDataBase {
             try statementBuilder.delete(tableName: D.tReport,
                                         primarykeyValue: serverCondition)
             
+            try statementBuilder.delete(tableName: D.tResource,
```

Comment:
@dhekra-rouatbi when a server is disconnected the resources should also be deleted from the device. In this case we should also remove the resource from the file manager, right? Should I create a separate method for this in the serversViewModel?

Reply:
I was thinking of something like this:

```
func deleteServer() {
        guard let serverId = self.currentServer?.id else { return  }

        if currentServer?.serverType == .uwazi {
            mainAppModel.vaultManager.tellaData?.deleteUwaziServer(serverId: serverId)
            
            return
        }
        
        mainAppModel.vaultManager.tellaData?.deleteServer(serverId: serverId)
        let resourcesId = mainAppModel.vaultManager.tellaData?.getResourcesByServerId(serverId: serverId)
        mainAppModel.vaultManager.deleteVaultFile(filesIds: [resourcesId])

    }
    ```
    
    what do you think?

Reply:
I think it's better to add  `var vaultManager : VaultManagerInterface?`  to `TellaData` and add this code to  `TellaData` (see `class VaultFilesManager : VaultFilesManagerInterface `)

```
    func deleteServer(server:Server) {
        guard let serverId = server.id else { return  }
        
        if server.serverType == .uwazi {
            deleteUwaziServer(serverId: serverId)
        } else {
            deleteServer(serverId: serverId)
        }
        
        let resourcesId =  getResourcesByServerId(serverId: serverId)
        mainAppModel.vaultManager.deleteVaultFile(filesIds: [resourcesId])
        deleteDownloadedResource(resourcesId: resourcesId)
    }


```

Reply:
great! what do you think of this?

```
func deleteServer(server: Server) {
        guard let serverId = server.id else { return }
        
        if server.serverType == .uwazi {
            deleteUwaziServer(serverId: serverId)
            return
        }
        
        deleteTellaServer(serverId: serverId)
        let resourcesId = getResourceByServerId(serverId: serverId)
        vaultManager?.deleteVaultFile(filesIds: resourcesId)
        
    }

  ```


since the resources will be deleted only if it's a tella server, I'm finishing the execution when is of type uwazi. So it only delete the server and it's done. 

then I'm getting the resourcesID and removing the vaultFiles. The resources in the database are being deleted in the same delete tella server query so we do the same as we do with reports.

Reply:
yeah you're right,  I just suggest doing it with `switch case` because it's more clear and we might have more servers in the future